src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -1257,6 +1292,18 @@ def individual_node_complexity(self) -> Dict[PlanNodeCategory, int]:
         # we add #set_operands - 1 additional operators in sql query
         return {PlanNodeCategory.SET_OPERATION: len(self.set_operands) - 1}
 
+    def replace_child(self, old_node, new_node) -> None:
+        if not self._is_deep_copied:
+            raise ValueError(
+                "replace child can only be called on a deep copied plan node."
+            )
+
+        if old_node not in self._nodes:
+            raise ValueError(
+                "old_node to be replaced is not found in the children nodes."
+            )
+        self._nodes = [node if node != old_node else new_node for node in self._nodes]
```

Comment:
you will also need to update the operant selectable and the pre and post actions

Reply:
don't we plan to reset the snowflake plan after the plan tree has been restructured. The query generation should handle that step right?